mbus/nats_handler.go

```diff
@@ -246,51 +271,44 @@ func (h *natsHandler) runUntilInterrupted() {
 	}
 }
 
-func (h *natsHandler) getConnectionInfo() (*yagnats.ConnectionInfo, error) {
+func (h *natsHandler) getConnectionInfo() (*ConnectionInfo, error) {
 	settings := h.settingsService.GetSettings()
 
-	natsURL, err := url.Parse(settings.GetMbusURL())
+	connInfo := new(ConnectionInfo)
+	connInfo.Addr = settings.GetMbusURL()
+	natsURL, err := url.Parse(connInfo.Addr)
 	if err != nil {
 		return nil, bosherr.WrapError(err, "Parsing Nats URL")
 	}
 
-	connInfo := new(yagnats.ConnectionInfo)
-	connInfo.Addr = natsURL.Host
+	hostSplit := strings.Split(natsURL.Host, ":")
+	connInfo.IP = hostSplit[0]
+
 
 	if settings.Env.IsNATSMutualTLSEnabled() {
```

Comment:
Not totally sure what the error will be, but looking at the `nats` job in the director, it started ignoring the username/password properties back in 2019, they are totally unused at this point so you can't even auth using username/password (unless you run a different nats server yourself I suppose)

Reply:
That means our current required authentication is `mTLS` and it shouldn't be optional, which is with the `if` statement. If we don't have a proper error to fail here we can fail later when trying to connect. I just thought that we can provide more information about the issue if we fail here, but I don't know what is the error you get from nats-io in this situation. It could be that it is also good enough.

Reply:
Good call, we removed the conditional and doubled checked what happens when you don't provide certs:
```
2021-11-18_19:55:40.12271 [main] 2021/11/18 19:55:40 ERROR - App run Running agent: Message Bus Handler: Starting nats handler: Getting connection info: Parsing certificate and private key: tls: failed to find any PEM data in certificate input
2021-11-18_19:55:40.12272 [main] 2021/11/18 19:55:40 ERROR - Agent exited with error: Running agent: Message Bus Handler: Starting nats handler: Getting connection info: Parsing certificate and private key: tls: failed to find any PEM data in certificate input
```
Which seems like a decent error to give somebody.